homeassistant/components/ezviz/update.py

```diff
@@ -0,0 +1,109 @@
+"""Support for EZVIZ sensors."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyezviz import HTTPError, PyEzvizError
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_COORDINATOR, DOMAIN
+from .coordinator import EzvizDataUpdateCoordinator
+from .entity import EzvizEntity
+
+PARALLEL_UPDATES = 1
+
+UPDATE_ENTITY_TYPES = UpdateEntityDescription(
+    key="version",
+    name="Firmware update",
+    device_class=UpdateDeviceClass.FIRMWARE,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EZVIZ sensors based on a config entry."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities(
+        EzvizUpdateEntity(coordinator, camera, sensor, UPDATE_ENTITY_TYPES)
+        for camera in coordinator.data
+        for sensor, value in coordinator.data[camera].items()
+        if sensor in UPDATE_ENTITY_TYPES.key
+        if value
+    )
+
+
+class EzvizUpdateEntity(EzvizEntity, UpdateEntity[EzvizDataUpdateCoordinator]):
```

Comment:
My bad:

```suggestion
class EzvizUpdateEntity(EzvizEntity, UpdateEntity):
```

The `EzvizEntity` already passes it to `CoordinatorEntity`, I messed up my previous suggestion. Sorry about that 🙏 

../Frenck

Reply:
No problem at all, I just undid the changes in your last suggestion.

Reply:
Well, yes/no. The coordinator definition shouldn't be needed, as it is already passed in as a generic in the `EzvizEntity`.

Reply:
No problem. I removed it, it passes mypy test locally on my machine so should be fine.